test/tcp_tunnel.js

```diff
@@ -56,7 +54,7 @@ describe('tcp_tunnel.createTunnel', () => {
     it('throws error if proxyUrl is not in correct format', () => {
         assert.throws(() => { createTunnel('socks://user:password@whatever.com:123', 'localhost:9000'); }, /must have the "http" protocol/);
         assert.throws(() => { createTunnel('socks5://user:password@whatever.com', 'localhost:9000'); }, /must contain hostname and port/);
-        assert.throws(() => { createTunnel('http://us%3Aer:password@whatever.com:345', 'localhost:9000'); }, /cannot contain the colon/);
+        // assert.throws(() => { createTunnel('http://us%3Aer:password@whatever.com:345', 'localhost:9000'); }, /cannot contain the colon/);
```

Comment:
we should probably add a TODO here and reenable the test later?

Reply:
>  ignore invalid : in username fields (if present, the upstream proxy will reject with not authorized error)

Reply:
i dont follow. you are commenting out an assertion, if its valid and you plan to uncomment it, please add a todo so we don't forget about that. if its invalid, why should we keep it commented instead of removing it?

Reply:
> ignore invalid ...

So the behavior will be different? I think we should keep track of all the breaking and possibly breaking changes so that we can compile a good changelog.